qiime2/sdk/usage.py

```diff
@@ -396,6 +413,17 @@ def _init_metadata_(self, ref, factory):
 
         return result
 
+    def _init_data_collection_(self, ref, collection_type, *records):
+        collection = []
+        for record in records:
+            result_type = type(record)
+            if not isinstance(record, ScopeRecord):
+                raise ValueError('Record (%r) returned a %s, expected a '
+                                 'ScopeRecord.' % (record, result_type))
```

Comment:
What do you think about putting this check up in the base class?

Reply:
Would you also want to move the other similar checks on the init data methods of `ExecutionUsage` into the base class?

Reply:
What checks did you have in mind? I'm not seeing any others, but please drop a link or two in to set my straight.

Reply:
You can disregard my comment.  I was confused.

Reply:
> I was confused.

story of my life.